src/pybind/mgr/cephadm/module.py

```diff
@@ -1701,7 +1701,9 @@ def _daemon_action(self,
             if daemon_spec.daemon_type != 'osd':
                 daemon_spec = self.cephadm_services[daemon_type_to_service(
                     daemon_spec.daemon_type)].prepare_create(daemon_spec)
-            return CephadmServe(self)._create_daemon(daemon_spec, reconfig=(action == 'reconfig'))
+            if action == 'reconfig':
+                return CephadmServe(self)._create_daemon(daemon_spec, reconfig=True)
+            return CephadmServe(self)._create_daemon(daemon_spec, redeploy=True)
```

Comment:
Idempotency. `CephadmServe._create_daemon` is supposed to be idempotent. That's why we don't pass `redeploy` to it. I don't think we should pass redeploy here.

Reply:
Changed it back to original